compiler/noirc_evaluator/src/ssa/ir/instruction/binary.rs

```diff
@@ -192,6 +192,22 @@ pub(crate) fn eval_constant_binary_op(
             let Some(rhs) = try_convert_field_element_to_signed_integer(rhs, bit_size) else {
                 return CouldNotEvaluate;
             };
+
+            // Because we always perform signed operations using i128, an operation like `-128_i8 / -1`
+            // will not overflow as it'll actually be done via `-128_i128 / -1`. Thus we need to
+            // manually check this specific case.
+            if matches!(operator, BinaryOp::Div | BinaryOp::Mod) && rhs == -1 {
+                assert!(bit_size < 128);
+                let min_value = -(1_i128 << (bit_size - 1));
```

Comment:
NB there is already a `two_pow_bit_size_minus_one` variable further down. I thought maybe it could be reused with some extra comments.

Reply:
Done!